powerline/segments/vim/plugin/ctrlp.py

```diff
@@ -0,0 +1,102 @@
+# vim:fileencoding=utf-8:noet
+from __future__ import (unicode_literals, division, absolute_import, print_function)
+
+try:
+	import vim
+except ImportError:
+	vim = object()
+
+from powerline.bindings.vim import create_ruby_dpowerline
+
+
+def initialize():
+	global initialized
+	if initialized:
+		return
+	initialized = True
+	vim.command('echom 3210')
+	vim.command('let g:ctrlp_status = {}')
+	vim.command(
+		'''
+		execute 'function! Ctrlp_status_main(focus, byfname, regex, prev, item, next, marked)
+			let g:ctrlp_status = {\
+"focus": a:focus,\
+"byfname": a:byfname,\
+"regex": a:regex,\
+"prev": a:prev,\
+"item": a:item,\
+"next": a:next,\
+"marked": a:marked,\
+}
+			return ""
+		endfunction'
+		'''
+	)
+	vim.command(
+		'''
+		execute 'function! Ctrlp_status_prog(str)
+			let g:ctrlp_status["prog"] = a:str
+			return ""
+		endfunction'
+		'''
+	)
+	vim.command(
+		'''
+		let g:ctrlp_status_func = {\
+			'main': 'Ctrlp_status_main',\
+			'prog': 'Ctrlp_status_prog',\
+		}
+		'''
+	)
```

Comment:
I have tested vimcommand.py file that looks like

```Python
vim.command(
		'''
		function! Ctrlp_status_main(focus, byfname, regex, prev, item, next, marked)
			let g:ctrlp_status = {\
				"focus": a:focus,\
				"byfname": a:byfname,\
				"regex": a:regex,\
				"prev": a:prev,\
				"item": a:item,\
				"next": a:next,\
				"marked": a:marked,\
			}
			return ""
		endfunction
		function! Ctrlp_status_prog(str)
			let g:ctrlp_status["prog"] = a:str
			return ""
		endfunction
		let g:ctrlp_status_func = {\
			'main': 'Ctrlp_status_main',\
			'prog': 'Ctrlp_status_prog',\
		}
		'''
	)
```

and it worked to define the first function fine (but for some reason not the second, so you can’t squash all the `vim.command` calls).

```Python
vim.command(
		'''
		function! Ctrlp_status_main(focus, byfname, regex, prev, item, next, marked)
			let g:ctrlp_status = {\
				"focus": a:focus,\
				"byfname": a:byfname,\
				"regex": a:regex,\
				"prev": a:prev,\
				"item": a:item,\
				"next": a:next,\
				"marked": a:marked,\
			}
			return ""
		endfunction
		'''
)
vim.command(
		'''
		function! Ctrlp_status_prog(str)
			let g:ctrlp_status["prog"] = a:str
			return ""
		endfunction
		'''
)
vim.command(
		'''
		let g:ctrlp_status_func = {\
			'main': 'Ctrlp_status_main',\
			'prog': 'Ctrlp_status_prog',\
		}
		'''
)
```

defined everything you need.